src/jailcheck/access.c

```diff
@@ -80,10 +80,13 @@ void access_setup(const char *directory) {
 	FILE *fp = fopen(test_file, "w");
 	if (!fp) {
 		printf("Warning: I cannot create test file in directory %s, skipping...\n", directory);
+		free(path);
+		free(test_file);
 		return;
 	}
```

Comment:
```suggestion
	if (!fp) {
		printf("Warning: I cannot create test file in directory %s, skipping...\n", directory);
		free(test_file);
		free(path);
		return;
	}
```

Free in reverse order.

Reply:
Done